app/components/editable_champ/editable_champ_component.rb

```diff
@@ -1,5 +1,48 @@
 class EditableChamp::EditableChampComponent < ApplicationComponent
-  def initialize(form:, champ:)
-    @form, @champ = form, champ
+  def initialize(form:, champ:, seen_at: nil)
+    @form, @champ, @seen_at = form, champ, seen_at
+  end
+
+  private
+
+  def has_label?(champ)
+    types_without_label = [TypeDeChamp.type_champs.fetch(:header_section), TypeDeChamp.type_champs.fetch(:explication)]
+    !types_without_label.include?(@champ.type_champ)
+  end
+
+  def component_class
+    "EditableChamp::#{@champ.type_champ.camelcase}Component".constantize
+  end
+
+  def html_options
+    {
+      class: "editable-champ-#{@champ.type_champ} #{@champ.visible? ? '' : 'hidden'}",
```

Comment:
Je tenterai bien ça (sans garantie)

```suggestion
      class: ["editable-champ-#{@champ.type_champ}", "hidden": @champ.visible?],
```

Reply:
marche pas, mais j'ai fait un truc un peu plus lisible